iconsMapper.ts

```diff
@@ -8,15 +23,66 @@ const mdiViewBox = {
 
 const remap = reMapIcons();
 
+/**
+ * Get the icon lookup from the icon string
+ * @param icon
+ * @returns IconLookup
+ */
+function getIconName(icon: string): IconLookup {
+  const prefix = icon.split('-', 1)[0];
+  const iconName = icon.slice(prefix.length + 1);
+
+  return {
+    prefix,
+    iconName,
+  };
+}
+
+/**
+ * Re-map the icons according to the mapping
+ * Unknown and null mapping is ignored - original MDI icon will be used
+ */
 function reMapIcons() {
   const result = {
     icons: { ...mdiIcons },
     viewBoxes: {},
+    stats: {},
   };
 
   for (const icon in result.icons) {
     // add the viewBox for the icon in order to build the template correctly
     result.viewBoxes[icon] = mdiViewBox;
+    result.stats[icon] = MappingStatus.ORIGINAL;
+    if (!mapping.icons.hasOwnProperty(icon)) {
```

Comment:
1. This check will bail out for unmapped icons too, making the next check that logs and collects "stats" redundant/no-op
    An unmapped icon would not be logged and not show up in the stats.
    I assume the purpose was to skip Javascript internal properties. They should not be present with `for..in` anyway.
    I think it is safe to remove this condition.
2. `hasOwnProperty()` is [deprecated][mdn-hasownprop], please use `Object.hasOwn()` instead

Altenative approach: make two diff (both ways) and one intersection of the keys of `result.icons` and `mapping.icons`, dump the differences (unmapped or unknown) and process the intersection.

[mdn-hasownprop]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn

Reply:
_"An unmapped icon would not be logged and not show up in the stats."_ - all stats are userful for building the contactsheet. To help the map person to find faster the map errors

Reply:
I understand what the stats are for.

However, the `hasOwnProperty()` and the following `!newIconName` are redundant because they both test for undefined field. 

I now see that the 2nd branch also tests for an empty string, in that case it's not redundant and the state is sensible. 

I'd propose to change the log message and maybe change to `newIconName === ''` to make this clearer or to test for a name pattern via a RegExp if this is about preventing malformed/unexpected icon names. Maybe we can even lookup the name in the library?